app/Jobs/PruneAccessFromAccessInactiveUsers.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Jobs;
+
+use App\Models\User;
+use Carbon\CarbonImmutable;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+
+class PruneAccessFromAccessInactiveUsers implements ShouldQueue
+{
+    use Dispatchable;
+    use InteractsWithQueue;
+    use Queueable;
+    use SerializesModels;
+
+    /**
+     * Create a new job instance.
+     */
+    public function __construct()
+    {
+        $this->queue = 'jedi';
+    }
+
+    /**
+     * Execute the job.
+     */
+    public function handle(): void
+    {
+        User::accessInactive(CarbonImmutable::now()->subDay())
+            ->where('is_service_account', '=', false)
+            ->where(static function (Builder $query): void {
+                $query->whereHas('roles')
+                    ->orWhereHas('permissions');
+            })
+            ->get()
+            ->filter(static fn (User $user): bool => ! $user->is_access_active)
```

Comment:
The check down here is to make sure the users are _still_ access-inactive. I can't easily chain the same scope on itself so this seemed like the least bad option.